meshtastic/node.py

```diff
@@ -1023,16 +1023,35 @@ def ensureSessionKey(self):
     def get_channels_with_hash(self):
         """Return a list of dicts with channel info and hash."""
         result = []
+        def format_preset_name(name):
+            # Convert name like MODEM_PRESET to ModemPreset
+            return ''.join(word.capitalize() for word in name.split('_'))
+
         if self.channels:
             for c in self.channels:
-                if c.settings and hasattr(c.settings, "name") and hasattr(c.settings, "psk"):
-                    hash_val = generate_channel_hash(c.settings.name, c.settings.psk)
-                else:
-                    hash_val = None
-                result.append({
-                    "index": c.index,
-                    "role": channel_pb2.Channel.Role.Name(c.role),
-                    "name": c.settings.name if c.settings and hasattr(c.settings, "name") else "",
+                # Ignore DISABLED channels
+                if channel_pb2.Channel.Role.Name(getattr(c, "role", 0)) == "DISABLED":
+                    continue
+                hash_val = None
+                name = ""
+                if getattr(c, "settings", None) is not None:
+                    if hasattr(c.settings, "name") and hasattr(c.settings, "psk"):
+                        hash_val = generate_channel_hash(c.settings.name, c.settings.psk)
+                    name = getattr(c.settings, "name", "")
+                # If PRIMARY and name is empty, use formatted preset name from localConfig.lora.modem_preset
+                if c.role == channel_pb2.Channel.Role.PRIMARY and not name:
```

Comment:
This is what I meant in discord about it not just being primary channels. This needs to be done for secondary ones too.

```suggestion
                if not name:
```